Karumi/Shot
# Update facebook dependency to 0.5.0

### :pushpin: References
* **Issue:** #35 #33 
* **Related pull-requests:** #20 

### :tophat: What is the goal?

* To increment the facebook screenshot testing library version to, at least, `0.5.0`.
* To be able to run screenshot tests in Android emulators with SDK version greater than 22

### How is it being implemented?

* Increase facebook library version to 0.5.0
* Include a library to be able to parse JSON because the new screenshot test library now uses that format to specify meta-information for each test.
* Replace XML parsing by JSON parsing when retrieving the height and width of the output screenshots.
* Increase android emulator versions to SDK 23 in order to test that the new feature is working.
* To lock the facebook screenshot testing library version to 0.5.0 in case the host application is also loading it. We are doing it so that we don't get incompatible versions installed.

### How can it be tested?

1. Run `./createEmulator.sh`. This will create a new emulator with SDK 23
2. Run `./startEmulator.sh`. This will start the newly created emulator
3. Run `./gradlew executeScreenshotTests` and verify that all tests are passing.
